### ISSUE TITLE
Train own model  还是出现batch size大于训练样本问题

### ISSUE THREAD

**Issue by xiaozi-ML** [MaybeShewill-CV/lanenet-lane-detection#159]
您好！我看了之前的问题，已经将training_data_example样本复制了3遍且修改了图片名。
[train.txt](https://github.com/MaybeShewill-CV/lanenet-lane-detection/files/2819984/train.txt)
[val.txt](https://github.com/MaybeShewill-CV/lanenet-lane-detection/files/2819986/val.txt)这两个文件也作了修改，但是还出现“Batch size不能大于样本的总数量”这样的错误，请问这个是什么原因？

**Comment by MaybeShewill-CV** [comment]
@xiaozi-ML 你可以下载tusimple数据集训练:)

**Comment by MaybeShewill-CV** [comment]
@xiaozi-ML 可以再检查你的训练数据准备。是不是没有图像数据或者train.txt文件有问题:)

**Comment by xiaozi-ML** [comment]
我把 batch size 改为1  就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
”，有点无奈，搞了2天了，一直都没解决。

**Comment by MaybeShewill-CV** [comment]
@xiaozi-ML 贴一下你的出错截图:)

**Comment by PhilWallace** [comment]
> 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> ”，有点无奈，搞了2天了，一直都没解决。

请问你解决了吗？我也遇到了同样的问题..

**Comment by PhilWallace** [comment]
> 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> ”，有点无奈，搞了2天了，一直都没解决。

解决了……是不细心，参数没改完

**Comment by xiaozi-ML** [comment]
> > 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> > ”，有点无奈，搞了2天了，一直都没解决。
> 
> 解决了……是不细心，参数没改完

请问你是怎么解决的？

**Comment by PhilWallace** [comment]
> 
> 
> > > 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> > > ”，有点无奈，搞了2天了，一直都没解决。
> > 
> > 
> > 解决了……是不细心，参数没改完
> 
> 请问你是怎么解决的？

config文件里有好几个batch size参数，我当时没改完。全部修改正确过后就可以训练了

**Comment by xiaozi-ML** [comment]
> > > > 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> > > > ”，有点无奈，搞了2天了，一直都没解决。
> > > 
> > > 
> > > 解决了……是不细心，参数没改完
> > 
> > 
> > 请问你是怎么解决的？
> 
> config文件里有好几个batch size参数，我当时没改完。全部修改正确过后就可以训练了
麻烦帮我看下，我一共改了3个batch size,还是报错：
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# @Time    : 18-1-31 上午11:21
# @Author  : Luo Yao
# @Site    : http://icode.baidu.com/repos/baidu/personal-code/Luoyao
# @File    : global_config.py
# @IDE: PyCharm Community Edition
"""
设置全局变量
"""
from easydict import EasyDict as edict

__C = edict()
# Consumers can get config by: from config import cfg

cfg = __C

# Train options
__C.TRAIN = edict()

# Set the shadownet training epochs
__C.TRAIN.EPOCHS = 200010
# Set the display step
__C.TRAIN.DISPLAY_STEP = 1
# Set the test display step during training process
__C.TRAIN.TEST_DISPLAY_STEP = 1000
# Set the momentum parameter of the optimizer
__C.TRAIN.MOMENTUM = 0.9
# Set the initial learning rate
__C.TRAIN.LEARNING_RATE = 0.0005
# Set the GPU resource used during training process
__C.TRAIN.GPU_MEMORY_FRACTION = 0.85
# Set the GPU allow growth parameter during tensorflow training process
__C.TRAIN.TF_ALLOW_GROWTH = True
# Set the shadownet training batch size
__C.TRAIN.BATCH_SIZE = 2

# Set the shadownet validation batch size
__C.TRAIN.VAL_BATCH_SIZE = 2
# Set the learning rate decay steps
__C.TRAIN.LR_DECAY_STEPS = 410000
# Set the learning rate decay rate
__C.TRAIN.LR_DECAY_RATE = 0.1
# Set the class numbers
__C.TRAIN.CLASSES_NUMS = 2
# Set the image height
__C.TRAIN.IMG_HEIGHT = 256
# Set the image width
__C.TRAIN.IMG_WIDTH = 256

# Test options
__C.TEST = edict()

# Set the GPU resource used during testing process
__C.TEST.GPU_MEMORY_FRACTION = 0.8
# Set the GPU allow growth parameter during tensorflow testing process
__C.TEST.TF_ALLOW_GROWTH = True
# Set the test batch size
__C.TEST.BATCH_SIZE = 2

**Comment by PhilWallace** [comment]
> 
> 
> > > > > 我把 batch size 改为1 就出现了这个错误“error: (-215:Assertion failed) !ssize.empty() in function 'cv::resize'
> > > > > ”，有点无奈，搞了2天了，一直都没解决。
> > > > 
> > > > 
> > > > 解决了……是不细心，参数没改完
> > > 
> > > 
> > > 请问你是怎么解决的？
> > 
> > 
> > config文件里有好几个batch size参数，我当时没改完。全部修改正确过后就可以训练了
> > 麻烦帮我看下，我一共改了3个batch size,还是报错：
> > #!/usr/bin/env python3
> 
> # -_- coding: utf-8 -_-
> # @time : 18-1-31 上午11:21
> # @author : Luo Yao
> # @Site : http://icode.baidu.com/repos/baidu/personal-code/Luoyao
> # @file : global_config.py
> # @ide: PyCharm Community Edition
> 
> """
> 设置全局变量
> """
> from easydict import EasyDict as edict
> 
> __C = edict()
> # Consumers can get config by: from config import cfg
> 
> cfg = __C
> # Train options
> 
> __C.TRAIN = edict()
> # Set the shadownet training epochs
> 
> __C.TRAIN.EPOCHS = 200010
> # Set the display step
> 
> __C.TRAIN.DISPLAY_STEP = 1
> # Set the test display step during training process
> 
> __C.TRAIN.TEST_DISPLAY_STEP = 1000
> # Set the momentum parameter of the optimizer
> 
> __C.TRAIN.MOMENTUM = 0.9
> # Set the initial learning rate
> 
> __C.TRAIN.LEARNING_RATE = 0.0005
> # Set the GPU resource used during training process
> 
> __C.TRAIN.GPU_MEMORY_FRACTION = 0.85
> # Set the GPU allow growth parameter during tensorflow training process
> 
> __C.TRAIN.TF_ALLOW_GROWTH = True
> # Set the shadownet training batch size
> 
> __C.TRAIN.BATCH_SIZE = 2
> # Set the shadownet validation batch size
> 
> __C.TRAIN.VAL_BATCH_SIZE = 2
> # Set the learning rate decay steps
> 
> __C.TRAIN.LR_DECAY_STEPS = 410000
> # Set the learning rate decay rate
> 
> __C.TRAIN.LR_DECAY_RATE = 0.1
> # Set the class numbers
> 
> __C.TRAIN.CLASSES_NUMS = 2
> # Set the image height
> 
> __C.TRAIN.IMG_HEIGHT = 256
> # Set the image width
> 
> __C.TRAIN.IMG_WIDTH = 256
> # Test options
> 
> __C.TEST = edict()
> # Set the GPU resource used during testing process
> 
> __C.TEST.GPU_MEMORY_FRACTION = 0.8
> # Set the GPU allow growth parameter during tensorflow testing process
> 
> __C.TEST.TF_ALLOW_GROWTH = True
> # Set the test batch size
> 
> __C.TEST.BATCH_SIZE = 2

都改过就可能是其他的问题了……关键你这里的数据得和你文件夹里的样本保持一致，再检查看看吧
